File: bench/00_torch_mm.py
import torch
import time
import os

batch_size = int(os.environ.get('BATCH_SIZE', "1000"))
device = os.environ.get("DEVICE", "cuda")

def mm ():
  x = torch.randn(batch_size, 784, device = device)
  w = torch.randn(784, 512, device = device)
  for i in range(ir):
    torch.mm(x, w).cpu()

    
ir = 1    
mm()
ir = int(os.environ.get('ITER', "1000"))

start_time = time.time()
mm()
print((time.time() - start_time))

File: bench/00_torch_mm.R
library(torch)

batch_size <- as.numeric(Sys.getenv("BATCH_SIZE", unset = "1000"))

w <- torch_randn(784, 512)
x <- torch_randn(batch_size, 784)

f <- function() {
  for (i in 1:iter) {
    torch_mm(x, w)
  }

  invisible(NULL)
}

iter <- 1
f()
iter <- as.numeric(Sys.getenv("ITER", unset = "1000"))


cat(system.time(f())[["elapsed"]])
